cmd/opampsupervisor/supervisor/commander/commander.go

```diff
@@ -102,17 +104,19 @@ func (c *Commander) watch() {
 
 	c.running.Store(0)
 	close(c.doneCh)
```

Comment:
This is probably fine as-is, but an alternative could be to turn this into a buffered channel with a capacity of 1, then clear the channel if there's anything in it at startup.